modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java

```diff
@@ -382,6 +385,11 @@ public void processOpts() {
             this.setConfigKey(additionalProperties.get(CONFIG_KEY).toString());
         }
 
+        if(additionalProperties.containsKey(CONFIG_KEY_FROM_CLASS_NAME)) {
+            this.setConfigKey(null);
```

Comment:
What if `configKeyFromClassName` is set to `false`? Then the expectation is to still use the value of `configKey`, but it's set to `null` here.

Reply:
I have updated this to be conditional on CONFIG_KEY being null. So that the CONFIG_KEY option takes precedence